docs/diagrams/developer-guide/ParserClasses.puml

```diff
@@ -0,0 +1,38 @@
+@startuml
+!include style.puml
+skinparam arrowThickness 1.1
+skinparam arrowColor LOGIC_COLOR_T4
+skinparam classBackgroundColor LOGIC_COLOR
+
+Class "{abstract}\nCommand" as Command
+Class XYZCommand
+
+package "Parser classes"{
+Class "<<interface>>\nParser" as Parser
+Class TrackOParser
+Class XYZCommandParser
+Class CliSyntax
+Class ParserUtil
+Class ArgumentMultimap
```

Comment:
I'm not sure if this is an issue but right now the `Parser` class diagram does not account for the `FlagTokenizer` at all. And I'm not sure whether it should be included because not all `CommandParser` have a dependency on `FlagTokenizer`

Reply:
Do you mean something like this?

![image](https://user-images.githubusercontent.com/97272807/200175404-7ec3ce51-5384-4d3c-850d-98ae7d301b58.png)

Reply:
Yep, there are some other classes related to this :/ 
* `Prefix` implements an `ArgumentToken` interface
* there is also a `Flag` class which implements an `ArgumentToken` interface 
* `FlagTokenizer` depends on `Flag`
* `ArgumentMultimap` contains `ArgumentToken` (I'm guessing this is an aggregation relationship, not a composition relationship because when `ArgumentMultimap` is destroyed, neither `Flag` nor `Prefix` will be destroyed because they are defined in the `CliSyntax` class) 
Can we just show this as a dependency relationship?
* I could not find any reference to `Prefix` in the `ParserUtil` class :/

Ah I'm so sorry that I'm only bringing this up now :( I modified the structure of the argument tokens to accommodate for flags when I was implementing `marko` 

I was thinking of something like this: 
![ParserClasses](https://user-images.githubusercontent.com/97423167/200178646-3f3eeb84-9d1b-4795-9bd2-84f530d8e195.png)

Reply:
To be honest I'm not very familiar with the parser components since I didn't implement `marko` nor `findo`. Would you like to just make the changes to the diagram above in your currently open DG pr? Then you can just replace whatever diagram I have currently during the merge.

Reply:
ok, that works! Thanks Enze :)